src/logging/DefaultLogger.ts

```diff
@@ -75,4 +76,42 @@ export class Logger {
     debug(message: string): void {
         Debug(`debug:${this.moduleName}`)(message);
     }
+
+    /**
+     * Log an object with pretty print formatting and also
+     * masking sensitive information if required
+     *
+     * @param id - Unique label.
+     * @param object - Object to be logged
+     */
+    logObject(logLevel: string, id: string, object: any, stringify?: true): void {
+        const objectToLogJson: string = stringify !== true ? object : JSON.stringify(object, null, 2);
+        const message = this.sanitize(objectToLogJson);
```

Comment:
I don't think we want to sanitize all objects by default here.  the issue is that a different implementation of logObject may not do it and then we are hosed.  Suggest hoist this to all the call sites of logObject... then the two problems (object logging vs sanitize) are kept orthogonal.

Reply:
You're right, hoisted `this.sanitize()` to all call sites of logObject